psycopg/psycopg/types/hstore.py

```diff
@@ -98,6 +140,43 @@ def load(self, data: Buffer) -> Hstore:
         return rv
 
 
+class BaseHstoreBinaryLoader(RecursiveLoader):
+    format = Format.BINARY
+    encoding: str
```

Comment:
As explained in the comment in `_register_hstore`, we can't make this a [sub]class property because the adapter in the global context would be wrong - it's applied to connections with potentially different encodings. The encoding should be an instance property.

```suggestion
class BaseHstoreBinaryLoader(Loader):
    format = Format.BINARY

    def __init__(self, oid: int, context: AdaptContext | None = None):
        super().__init__(oid, context)
        enc = conn_encoding(self.connection)
        self.encoding = enc if enc != "ascii" else "utf-8"
```

It is not necessary to derive from `RecursiveLoader` if the object doesn't use `_tx`.